### PR TITLE
Index name restrictions in MerkleDB

### PR DESCRIPTION
**Pull request by pavel-mukhanov**
## Overview
Add some documentation and revert index name restrictions.

---

### Definition of Done

- [x] There are no TODOs left in the merged code
- [x] Change is covered by automated tests
- [x] Benchmark results are attached (if applicable)
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: components/merkledb/src/views/tests.rs [hunk 1]
```diff
@@ -847,3 +850,62 @@ fn multiple_patch() {
...
+    assert!(!check_valid_name("1in!dex_Namez"));
+    assert!(!check_valid_name("space name"));
+    assert!(!check_valid_name(" space "));
+}
+
+#[test]
+fn valid_name_for_url() {
+    assert_valid_name_url("service_name");
+    assert_valid_name_url("service_name\\");
+    assert_valid_name_url("sarvice_name");
```
Comment (aleksuss) [inline components/merkledb/src/views/tests.rs]:
```suggestion
    assert_valid_name_url("service_name");
```

File: components/merkledb/src/views/tests.rs [hunk 2]
```diff
@@ -847,3 +850,62 @@ fn multiple_patch() {
...
 }
+
+#[test]
+fn valid_index_name() {
+    assert!(check_valid_name("index_name"));
+    assert!(check_valid_name("_index_name"));
+    assert!(check_valid_name("AinDex_name_"));
+    assert!(check_valid_name("core.index_name1Z"));
+    assert!(check_valid_name("configuration.indeX_1namE"));
+    assert!(check_valid_name("1index_Namez"));
```
Comment (aleksuss) [inline components/merkledb/src/views/tests.rs]: To make spell checker happy should be replaced for: 
```suggestion
    assert!(check_valid_name("1index_Name"));
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -60,6 +60,9 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 #### exonum-merkledb
 
+- Added restrictions to index names. Allowable characters in index name: ASCII
```
Comment (skletsun) [inline CHANGELOG.md]: Allowable -> Allowed?

File: components/merkledb/src/db.rs [hunk 4]
```diff
@@ -204,6 +204,10 @@ impl WorkingPatch {
     }
 
     /// Returns a mutable reference to the changes corresponding to a certain index.
+    ///
+    /// # Panics
+    ///
+    /// If two indices with the same `address` exist in `Fork` that uses this patch.
```
Comment (skletsun) [inline components/merkledb/src/db.rs]: It is not possible for two indices with the same `address` to exist within single `Fork`. Wouldn't it be better to reword like "If an index with the `address` already exists in ..."?

**Pull request merged**